### DISCUSSION TITLE
API not working in EU

### DISCUSSION THREAD

**Discussion by Mqtth3w** [ping/instagram_private_api#433] [Q&A]
The API doesn't work in EU because Instagram requires a client up to date

**Comment by Mqtth3w** [comment]
Solved. Just update the user agent. https://github.com/ping/instagram_private_api/blob/master/instagram_private_api/constants.py. 
Use a non-European IP address.

**Reply by Perinozzi-Mattia** [reply]
How?

**Reply by StanisLoveSid** [reply]
Could you elaborate please? You updated user agent directly in the library dependancy? What exactly should be updated in the user agent? Could you provide exact values? Thank you for your time!

**Reply by Mqtth3w** [reply]
Just look at the user agent file. All data are too old, update the library locally in your machine. I think It's enough updating the app data. In your terminal do "pip show instagram_private_api" you will see the email of the library owner, if you all ask him to update the library maybe he/she will update.

**Reply by StanisLoveSid** [reply]
I've updated the constants with new values [as shown here](https://github.com/dilame/instagram-private-api/blob/3a153c4e807049e95d72af67e316d432c2fd1577/src/core/constants.ts) but after this I get 
`ClientError Unable to get csrf from prelogin. (Code: 0, Response: {"status":"ok"})`
I guess this application should be updated in general because instagram api was changed a lot.

**Reply by Mxlan2711** [reply]
Any news on this ?

**Reply by clmtdq** [reply]
No
